src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -2048,16 +2048,16 @@ MavlinkReceiver::handle_message_manual_control(mavlink_message_t *msg)
 		_rc_pub.publish(rc);
 
 	} else {
-		manual_control_setpoint_s manual{};
+		manual_control_setpoint_s manual_control_setpoint{};
 
-		manual.timestamp = hrt_absolute_time();
-		manual.x = man.x / 1000.0f;
-		manual.y = man.y / 1000.0f;
-		manual.r = man.r / 1000.0f;
-		manual.z = man.z / 1000.0f;
-		manual.data_source = manual_control_setpoint_s::SOURCE_MAVLINK_0 + _mavlink->get_instance_id();
+		manual_control_setpoint.timestamp = hrt_absolute_time();
+		manual_control_setpoint.x = man.x / 1e3f;
+		manual_control_setpoint.y = man.y / 1e3f;
+		manual_control_setpoint.z = (man.z / 1e3f * 2.f) - 1.f;
```

Comment:
Likewise, I thought the point of this was to avoid the remapping? I realize we might have to adjust expectations on the QGC side. https://mavlink.io/en/messages/common.html#MANUAL_CONTROL

Reply:
Yes, you're right. Let's get this straight and check if it's breaking anything.

Reply:
I switched it any made a comment in the commit message. I can't currently test functionality together with QGC since SITL startup scripts are still broken and I'll have to fix that first: https://github.com/PX4/Firmware/issues/15795

Reply:
I got it working temporarily by renaming all `source` arbitrarily with `so` like described in https://github.com/PX4/Firmware/issues/15795#issuecomment-699494925
QGC still sends [0,1]. This might be a good moment to introduce general checks for stick centering and ranges to make sure nothing bad happens.

Reply:
I swtiched it back temporarily to keep compatibility to QGC for the moment I probably can't rebase this pr until QGC isd changed. I can look into making a pr for QGC but I'm not sure how we should deal with versioning.